### PR TITLE
Add Document Codes step

### PR DESCRIPTION
**Pull request by tavy88**
### Jira link

https://transformuk.atlassian.net/browse/HOTT-841

### What?

I have added/removed/altered:

- [x] Add Document Codes step to the duty calculator
- [x] Hide the step behind a feature flag

### Why?

I am doing this because:

- Business requirement - needed for duty calculation later on

Screen Shot 2021-07-22 at 10 59 07

### CODE REVIEW

File: spec/models/user_session_spec.rb [hunk 1]
```diff
@@ -399,6 +435,72 @@
...
+
+    before do
+      user_session.document_code_uk = value
+    end
+
+    it 'stores the hash on the session' do
+      expect(session['answers'][Steps::DocumentCode.id]).to eq(expected_value)
+    end
+
+    it 'merges new additional codes to the existing ones' do
```
Comment (tavy88) [inline spec/models/user_session_spec.rb:464]: A typo here needs to say document codes here.

File: spec/models/user_session_spec.rb [hunk 2]
```diff
@@ -399,6 +435,72 @@
...
+
+    before do
+      user_session.document_code_xi = value
+    end
+
+    it 'stores the hash on the session' do
+      expect(session['answers'][Steps::DocumentCode.id]).to eq(expected_value)
+    end
+
+    it 'merges new additional codes to the existing ones' do
```
Comment (tavy88) [inline spec/models/user_session_spec.rb:497]: Typo needs to say document codes.

File: app/helpers/commodity_helper.rb [hunk 3]
```diff
@@ -48,7 +64,11 @@ def applicable_excise_additional_codes?
   end
 
   def applicable_measure_type_ids
-    @applicable_measure_type_ids ||= applicable_additional_codes.flat_map { |_service, additional_codes| additional_codes.keys }.uniq
+    applicable_additional_codes.flat_map { |_service, additional_codes| additional_codes.keys }.uniq
```
Comment (willfish) [inline app/helpers/commodity_helper.rb:67]: I guess we're happy to stop memoizing?

File: spec/models/user_session_spec.rb [hunk 4]
```diff
@@ -643,22 +759,6 @@
...
-      it { expect(user_session.additional_codes).to eq('2340, 2600, 2340, 2600') }
-    end
-
-    context 'when additional code answers have not been stored' do
-      subject(:user_session) { build(:user_session, :with_commodity_information) }
-
-      it { expect(user_session.additional_codes).to eq('') }
-    end
-  end
-
```
Comment (tavy88) [inline spec/models/user_session_spec.rb:661]: To move the method from user_session to confirmation_decorator.

**Review by jebw — APPROVED** [review]
LGTM :+1:

**Review by jebw — APPROVED** [review]
LGTM :+1:

**Pull request merged**